Sample Applications/ExpenseIt/ExpenseItDemo/CreateExpenseReportDialogBox.cs

```diff
@@ -19,14 +19,22 @@ public partial class CreateExpenseReportDialogBox : Window
         public CreateExpenseReportDialogBox()
         {
             InitializeComponent();
+
+            var app = Application.Current;
+            var expenseReport = (ExpenseReport) app.FindResource("ExpenseData");
+
+            if (expenseReport != null && !expenseReport.Initialized)
+            {
+                expenseReport.InitializeItems();
+            }
```

Comment:
This app does a lot of things a real app wouldn't do:

1. operate on a singleton ExpenseReport
2. pre-load that report with real LineItems
3. do the pre-loading in markup

This brought on the "change an original item" bug, because markup calls IList.Add and thus avoids the override of IList<LineItem>.Add that adds the PropertyChanged listener.  I hate to see so much code devoted to a problem that wouldn't arise in a real app.  But it should be solved, so that people following the sample aren't led astray.

Let's make the solution smaller:

1.  Don't expose an Initialized property
2. Rename InitializeItems to EnsureInitialized, and have it check _initialized to do the work only once
3. Replace 25-29 by  expenseReport?.EnsureInitialized();
4. Add a comment in EnsureInitialized explaining why it's needed, and why a real app wouldn't need it.

Reply:
On further thought, I think the app simply overrides the wrong method.  Instead of overriding Add, try overriding [InsertItem](https://docs.microsoft.com/en-us/dotnet/api/system.collections.objectmodel.observablecollection-1.insertitem?view=net-5.0).  This should be called even in the markup case, so you can get rid of all the Initialize code you added.